app/main.py

```diff
@@ -1,3 +1,44 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+
+from app.customer import Customer
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as f:
+        data = json.load(f)
+
+    fuel_price = data.get("FUEL_PRICE")
+
+    customers = [Customer(**customer) for customer in data["customers"]]
+    shops = [Shop(**shop) for shop in data["shops"]]
+
+    for customer in customers:
+        print(f"{customer.name} has {customer.money} dollars")
+        dict_shops_costs = {}
+        for shop in shops:
+            print(f"{customer.name}'s trip to the {shop.name}"
+                  f" costs {customer.total_costs(shop, fuel_price)}")
+            dict_shops_costs.update(
+                {shop.name: customer.total_costs(
+                    shop, fuel_price
+                )
+                }
+            )
+        min_cost_shop = min(
+            shops,
+            key=lambda shop: customer.total_costs(shop, fuel_price),
+            default=None)
```

Comment:
In case there are no arguments on the same line with an opening parenthesis, your closing parenthesis should not have any arguments on the same line as well. Please, resolve this issue here and in the other places as well